public/app/plugins/panel/geomap/utils/tootltip.ts

```diff
@@ -96,7 +96,10 @@ export const pointerMoveListener = (evt: MapBrowserEvent<MouseEvent>, panel: Geo
   panel.hoverPayload.layers = layers.length ? layers : undefined;
   panel.props.eventBus.publish(panel.hoverEvent);
 
-  panel.setState({ ttip: { ...hoverPayload } });
+  // This check optimizes Geomap panel re-render behavior (without it, Geomap renders on every mouse move event)
+  if (panel.state.ttip === undefined || panel.state.ttip?.layers !== hoverPayload.layers || hoverPayload.layers) {
```

Comment:
Does this catch the edge case where the mouse can move directly from one tooltip to another?

Reply:
Not 100% sure though in that case the layers object should be different so it should trigger a re-render 🤷🏻‍♂️

Reply:
Yes it appears to work great with for this edge case :) 

https://github.com/grafana/grafana/assets/22381771/1170359d-cdcc-4125-8828-800b52a863dc